File: Poo Youtube Kotlin/src/main/kotlin/Pessoa.kt
abstract class Pessoa(Nome: String, Idade: Int, Sexo: String) {
    //Atributos
    var nome = Nome
        private set
    var idade = Idade
        private set
    var sexo: String
        private set
    var exp: Int = 0
        private set
    init {
        if (Sexo.trim().equals("M", true) || Sexo.trim().equals("F", true)) {
            this.sexo = Sexo.capitalize()
        }else{
            throw IllegalArgumentException("Digite M ou F")
        }
    }
    //Metodos publicos
    fun ganharExperiencia(Quantidade: Int){
        this.exp += Quantidade
    }
    fun detalhespess(){
         println("Nome: $nome, $idade anos, $sexo, Exp: $exp")
    }
}

File: POO Youtube Python/Pessoa.py
from abc import ABC


class Pessoa(ABC):
    # Atributos
    __exp = 0

    # Construtor
    def __init__(self, Nome: str, Idade: int, Sexo: str):
        if Sexo.upper().strip() not in ['M', 'F']:
            raise TypeError('Digite "M" ou "F"')
        else:
            self.sexo = Sexo
        self.nome = Nome
        self.idade = Idade

    # Metodos publicos
    def ganharexp(self, Experiencia: int):
        self.__exp += Experiencia

    # Metodos especiais
    @property
    def exp(self):
        return self.__exp
